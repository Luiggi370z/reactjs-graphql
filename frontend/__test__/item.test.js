import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
	id: 'ABC123',
	title: 'A cool item',
	price: 5000,
	description: 'This item',
	image: 'dog.jpg',
	largeImage: 'largedog.jpg'
}

describe('Item', () => {
	it('render and matches the snapshot', () => {
		const wrapper = shallow(<ItemComponent item={fakeItem} />)

		expect(toJSON(wrapper)).toMatchSnapshot()
	})
})

// describe('<Item/>', () => {
// 	it('renders image properly', () => {
// 		const wrapper = shallow(<ItemComponent item={fakeItem} />)
// 		const img = wrapper.find('img')

// 		expect(img.props().src).toBe(fakeItem.image)
// 		expect(img.props().alt).toBe(fakeItem.title)
// 	})

// 	it('renders and displays properly', () => {
// 		const wrapper = shallow(<ItemComponent item={fakeItem} />)
// 		const PriceTag = wrapper.find('PriceTag')

// 		expect(PriceTag.children().text()).toBe('$50')
// 		expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
// 	})

// 	it('renders out the buttons properly', () => {
// 		const wrapper = shallow(<ItemComponent item={fakeItem} />)
// 		const buttonList = wrapper.find('.buttonList')

// 		expect(buttonList.children()).toHaveLength(3)
// 		expect(buttonList.find('Link')).toHaveLength(1)
// 		expect(buttonList.find('AddToCart')).toBeTruthy()
// 		expect(buttonList.find('DeleteItem')).toBeTruthy()
// 	})
// })
