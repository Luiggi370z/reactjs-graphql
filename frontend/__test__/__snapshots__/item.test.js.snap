// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    >
      Add to Cart
    </AddToCart>
    <DeleteItem
      id="ABC123"
    >
      Delete
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
